aio/src/app/custom-elements/code/code.component.ts

```diff
@@ -111,12 +112,13 @@ export class CodeComponent implements OnChanges {
     const leftAlignedCode = leftAlign(this.code);
     this.setCodeHtml(leftAlignedCode); // start with unformatted code
     this.codeText = this.getCodeText(); // store the unformatted code as text (for copying)
-
-    this.pretty
+    const skipPrettify = of(undefined);
+    const prettifyCode = this.pretty
         .formatCode(leftAlignedCode, this.language, this.getLinenums())
-        .pipe(tap(() => this.codeFormatted.emit()))
-        .subscribe(c => this.setCodeHtml(c), () => { /* ignore failure to format */ }
-    );
+        .pipe(tap(formattedCode => this.setCodeHtml(formattedCode)));
+
+    iif(() => this.language === 'none', skipPrettify, prettifyCode)
```

Comment:
I probably proposed using `iif()` here, but looking at it again, since we are subscribing to the Observable ourselves, there is no need for `iif()` here. This could just be:

```suggestion
    (this.language === 'none' ? skipPrettify : prettifyCode)
```

Reply:
When I make this change, the next line causes issues. Should I also remove it (I think yes...?)? It's this one:

```
.subscribe(() => this.codeFormatted.emit(), () => { /* ignore failure to format */ });

```

Reply:
No, the following line should not be removed afaict. What kinds of issues are you getting?

Reply:
I pushed a fixup commit with a fix.